File: python/hamming/hamming.py
def distance(s1, s2):
    seq1 = list(s1)
    seq2 = list(s2)

    if len(seq1) is not len(seq2):
        raise ValueError()

    distance = 0
    for idx, c in enumerate(seq1):
        if c is not seq2[idx]:
            distance += 1
    
    return distance

File: typescript/hamming/hamming.ts
export default class Hamming{
  compute(a ='', b = ''){
    if(a.length != b.length){
      throw Error('DNA strands must be of equal length.')
    }
    
    var distance = 0;
    for (var idx = 0; idx < a.length; idx++) {
      if(a[idx]!=b[idx]){
        distance+=1;
      }
    }
    return distance;
  }
}